newline-after-import with decorators

lukekarrys opened the pull request:
This fixes the crash outlined in #592, when using a `require` statement inside a decorator.

While I was in there I also noticed that `newline-after-import` wasn't properly reporting if the next line was a decorator. I added this so it would be properly reported, but I can remove that if decorator support isn't something you want to add (I know it's still experimental).

jfmengels commented:
@lukekarrys Awesome! Thanks a lot for this! Also thanks for the great report, which I did not have time to look at until now.

I'm not familiar with the decorators syntax, but this looks good to me.

As for supporting decorators... It's currently at stage 2, which is not the safest stage but means it will *likely* make it in the spec. I think the way this is implemented it should be fine even if it doesn't make it into the spec, or if it changes a bit. I'm fine with this at the moment. @benmosher?

lukekarrys commented:
I rebased this against `master` and fixed a conflict in `CHANGELOG.md`. I can remove the changelog commit too if that's something that you normally do before releases.

benmosher commented:
Nope, I'm cool with decorators. Maybe not always all of stage 2, but definitely decorators. 😎
